README.md

```diff
@@ -130,6 +130,53 @@ data List = (List.cons head tail) | (List.nil)
 ListEx2 = (List.cons 1 (List.cons 2 (List.cons 3 List.nil)))
 ```
 
+Match different kinds of terms, both matches are equivalent:
```

Comment:
"It's possible to pattern match on different kinds of terms. These three forms are equivalent" or similar